### PR TITLE
WIP: Create unassigned requests scope, and tests

### PR DESCRIPTION
**Pull request by eduardoj**
Co-authored-by: Dany Marcoux <dmarcoux@suse.com>
Co-authored-by: Saray Cabrera Padrón <scabrerapadron@suse.de>
Co-authored-by: Victor Pereira <vpereira@suse.com>

### CODE REVIEW

File: src/api/db/structure.sql [hunk 1]
```diff
@@ -1130,7 +1138,7 @@ CREATE TABLE `staging_workflows` (
   `updated_at` datetime NOT NULL,
   PRIMARY KEY (`id`),
   UNIQUE KEY `index_staging_workflows_on_project_id` (`project_id`)
-) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```
Comment (coolo) [inline src/api/db/structure.sql]: This looks wrong - and unrelated

File: src/api/app/models/project.rb [hunk 2]
```diff
@@ -79,6 +79,7 @@ def autocomplete(search)
 
   has_one :staging, class_name: 'StagingWorkflow', inverse_of: :project
   belongs_to :staging_workflow, inverse_of: :staging_projects
+  has_and_belongs_to_many :bs_requests
```
Comment (coolo) [inline src/api/app/models/project.rb]: This really deserves a more specific term - projects can have tons of relations to requests. Source action, target action, review, ... - but this is purely about selected into a staging project?

File: src/api/app/models/staging_workflow.rb [hunk 3]
```diff
@@ -1,4 +1,25 @@
...
+  validates :project_id, presence: true
+
+  def unassigned_requests
+    project_requests - staging_projects_requests - ignored_requests
+  end
+
+  private
+
+  def project_requests
+    project.bs_requests
```
Comment (coolo) [inline src/api/app/models/staging_workflow.rb]: what is supposed to fill this?

File: src/api/db/structure.sql (hunk 1 above)
Comment (dmarcoux) [reply src/api/db/structure.sql]: @eduardoj You must be using an old database.yml.

Comment (eduardoj) [reply src/api/db/structure.sql]: Fixed.

File: src/api/app/models/project.rb [hunk 4]
```diff
@@ -79,6 +79,7 @@ def autocomplete(search)
 
   has_one :staging, class_name: 'StagingWorkflow', inverse_of: :project
   belongs_to :staging_workflow, inverse_of: :staging_projects
+  has_many :staged_requests
```
Comment (coolo) [inline src/api/app/models/project.rb]: Now we're talking. So are these the requests in the target project that are staged or are these the requests selected into a staging project?

Comment (eduardoj) [reply src/api/app/models/project.rb]: @coolo:

> Now we're talking. So are these the requests in the target project that are staged or are these the requests selected into a staging project?

The first, the requests in the target project that are staged.

Anyway, this is still not finished. Marking the pull request as WIP.

File: src/api/db/migrate/20181011083046_create_staged_requests.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class CreateStagedRequests < ActiveRecord::Migration[5.2]
+  def change
+    # rubocop:disable Rails/CreateTableWithTimestamps
+    create_table :staged_requests, id: false do |t|
+      t.belongs_to :bs_request, index: true, type: :integer
+      t.belongs_to :project, index: true, type: :integer
+      t.index [:bs_request_id, :project_id], unique: true
```
Comment (coolo) [inline src/api/db/migrate/20181011083046_create_staged_requests.rb]: if I may suggest an alternative again as discussion point:

```
t.belongs_to :bs_request
# technically duplicated info but makes lookup much more feasible
t.integer :target_project_id, null: false
# constraint: staged_into_project_id one of target_project's staging_workflow projects
t.integer :staged_into_project_id, null: true
t.boolean :ignored, default: false

Comment (coolo) [reply src/api/db/migrate/20181011083046_create_staged_requests.rb]: Oh wait. ignored must be a string.

**Comment by eduardoj** [comment]
Superseded by #6097.

**Pull request closed without merging**